### PR TITLE
macos: target min osx sdk is 10.7

### PR DESCRIPTION
**Pull request by obastemur**
Some of the embedders (i.e. node-chakracore targets an earlier version of SDK)

### CODE REVIEW

**Comment by Fishrock123** [comment]
@obastemur I now get a slightly different error output, but it's still about the same length: https://gist.github.com/Fishrock123/c9dba8ff7d6c26e296d1fa7d6846b176

**Comment by obastemur** [comment]
The rest of the errors looks like c++ exceptions related. Have you disabled c++ exceptions ? or do you have chance to use shared lib instead? (remove `--static` from build args)

**Comment by Fishrock123** [comment]
> Have you disabled c++ exceptions ? 

Not sure if I can.. will follow up with Rust people

> or do you have chance to use shared lib instead?

Hmmm, so is that a dynamic lib then? Sorry I am still newer to some of this. Dx

Doesn't seem to help much though, even if I specify to link dynamically or statically without building with `--static`...

**Comment by obastemur** [comment]
Okay, thanks for the details. 

This PR does what it supposed to do. For the rest; we can continue on #1492 
If you can share some details on how to repro things, I may take a look at closer.

File: pal/inc/safemath.h [hunk 1]
```diff
@@ -156,717 +167,6 @@ inline bool DoubleFitsInIntType(double val)
...
-//      if( !ClrSafeInt<UINT32>::multiply( elementCount, sizeof(void*), tmp ) ||
-//              !ClrSafeInt<UINT32>::addition( tmp, headerSize, bufSize ) ) 
-//      { <overflow-error> }
-//      else { use bufSize }
-//      
-//-----------------------------------------------------------------------------
-// TODO: Any way to prevent unintended instantiations?  This is only designed to
-//  work with unsigned integral types (signed types will work but we probably 
-//  don't need signed support).
-template<typename T> class ClrSafeInt
```
Comment (dilijev) [inline pal/inc/safemath.h]: Was it necessary to remove ClrSafeInt because of SDK versions or was this a drive-by cleanup?

**Comment by dilijev** [comment]
LGTM

File: pal/inc/safemath.h (hunk 1 above)
Comment (obastemur) [inline pal/inc/safemath.h]: both cleanup and some newer SDK stuff used

**Comment by obastemur** [comment]
@dilijev Thanks for the review

**Comment by dilijev** [comment]
Side note, the wiki currently says we target 10.9+ (https://github.com/Microsoft/ChakraCore/wiki/Building-ChakraCore#os-x). Should we revise to 10.7+?

**Comment by obastemur** [comment]
@dilijev no we shouldn't. I'm not sure the compiler support etc. we need are available on 10.7.. SDK and actual OS are kinda separate things here.

**Comment by dilijev** [comment]
@obastemur Got it, thanks.
